### PR TITLE
Brains and brain-heads no longer destroyed by recycler

### LINKED ISSUE #21505: Heads destroyed by recycler
I think there's already an issue for this but too lazy to look it up. Detached heads get destroyed by a recycler with safety mode on.

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #21505

:cl: Cyberboss
fix: Brains and heads with brains trigger the emergency stop on the recycler
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
:-1: It's a feature

**Comment by lzimann** [comment]
hugbox

**Comment by KingPhilipIII** [comment]
Down with hugbox.

**Comment by Cyberboss** [comment]
Jeez, how hard is it to emag it/space the brain?

**Comment by KingPhilipIII** [comment]
If it's not that hard to dispose of it, then why bother changing this?

**Comment by Cyberboss** [comment]
If you close this, close the issue as well because thats all i care about

File: code/game/machinery/recycler.dm [hunk 1]
```diff
@@ -116,9 +116,12 @@ var/const/SAFETY_COOLDOWN = 100
 
 	for(var/i in to_eat)
 		var/atom/movable/AM = i
-		if(isliving(AM))
+		var/obj/item/bodypart/head/as_head = AM
+		var/brain_holder = istype(AM, /obj/item/organ/brain) || as_head.brain
```
Comment (RemieRichards) [inline code/game/machinery/recycler.dm]: there is NO guarantee here that ```as_head``` will he a head.
meaning there's no guarantee it'll have a ```.brain``` var.
meaning it'll runtime.

Comment (Cyberboss) [inline code/game/machinery/recycler.dm]: right i forgot to add a null check

Comment (RemieRichards) [inline code/game/machinery/recycler.dm]: No, you forgot to check that it's a head at all.

File: code/game/machinery/recycler.dm [hunk 2]
```diff
@@ -116,9 +116,12 @@ var/const/SAFETY_COOLDOWN = 100
 
 	for(var/i in to_eat)
 		var/atom/movable/AM = i
-		if(isliving(AM))
+		var/obj/item/bodypart/head/as_head = AM
+		var/brain_holder = istype(AM, /obj/item/organ/brain) || (as_head && as_head.brain)
```
Comment (Cyberboss) [inline code/game/machinery/recycler.dm]: Huh, TIL

**Comment by anconfuzedrock** [comment]
>activate disposal unit
>turn off pump
>load pump with brain

jesus, its not like its hard to get rid of small items.

**Comment by Anonmare** [comment]
Couldn't you just make recyclers hackable to turn off the safeties with the emag doing it in a single stroke? It's not like there aren't other ways to dispose of brains.

**Comment by Cyberboss** [comment]
Maybe but thats a different PR

**Pull request merged**